### DISCUSSION TITLE
Connection Pool

### DISCUSSION THREAD

**Discussion by ToTru** [psycopg/psycopg#655] [Q&A]
When a broken connection from a pool is recognised as Not working, during
- connection
- cursor
- execute?

Knowing this check() could be executed in the except block to refresh them.
But I need to know. Any Knowledge here?

**Comment by dvarrazzo** [comment]
Documentation is [here](https://www.psycopg.org/psycopg3/docs/advanced/pool.html#connection-quality).

Short answer is: on `execute()`. However I am considering adding an optional check callback on `connection()`, because a lot of people have the expectation that they can use the pool together with databases that close idle connections (for instance using [`idle_session_timeout`](https://www.psycopg.org/psycopg3/docs/advanced/pool.html#pool-and-idle-session-timeout-setting) or with postgres-like vendor pooling connections and actively closing inactive ones.

**Reply by dvarrazzo** [reply]
Just opened #656 to work on this idea.

**Comment by ToTru** [comment]
Thanks, this was what I need. To prevent polling I will call the check() method in case execute() fails.
